ambari-server/src/main/java/org/apache/ambari/server/state/ServiceGroup.java

```diff
@@ -46,6 +46,10 @@
 
   Set<ServiceGroupDependencyResponse> getServiceGroupDependencyResponses();
 
+  String getStackId();
+
+  void setStackId(String stackId);
```

Comment:
Add documentation here. "StackID" can mean anything - it should be explicitly stated that a Stack ID represents a specific mpack version: MPackName-version

So, for example:
Foo-1.0.0.0-b25
Bar-3.1.0.4-b2

Reply:
In ambari code, StackId is widely used as the combination of stackName-stackVersion, there has a class StackId is for this, so I do not think it is caused confusion.

Reply:
Why not use `StackId` then instead of `String`?

Reply:
In Ambari < 3.0, the Stack ID was typically a 2-digit umbrella, like "HDP 2.6". It did not represent unique versions, like HDP 2.6.0.0-1234. We're changing this now so that Stack ID refers to the unique, fully-qualified version. I guess that's why it's a bit confusing to me.